packages/teleport-plugin-common/src/builders/style-builders.ts

```diff
@@ -146,11 +157,19 @@ export const generateStylesFromStyleSetDefinitions = (
       if (styleRef.type === 'element-state') {
         if (type === 'reusable-component-style-map') {
           cssMap.unshift(
-            createCSSClassWithSelector(name, `&:${styleRef.meta.state}`, collecedMediaStyles)
+            createCSSClassWithSelector(
+              name,
+              `&${subselectors || ''}:${styleRef.meta.state}`,
```

Comment:
Can we try using this approach for selectors in react-jss
https://cssinjs.org/jss-plugin-nested?v=v10.6.0